src/main/ngapp/accio/src/app/editor/mask.directive.ts

```diff
@@ -10,41 +11,127 @@ export class MaskDirective {
   @Input() originalImageData: ImageData;
   @Input() scale: number;
   @Input() tolerance: number;
-  @Input() disable: boolean;
+  @Input() disableFloodFill: boolean;
+  @Input() tool: MaskTool;
 
   @Output() newMaskEvent = new EventEmitter<Set<number>>();
+  @Output() newPaintEvent = new EventEmitter<number>();
+
+  //  Set containing pixels converted to their red index in ImageData. Used for paint and scribble
+  paintPixels: Set<number>;
+  mouseDown: boolean = false;
+  //  Determines if the user moved after they clicked, and does scribble fill instead of flood. 
+  scribbleFill: boolean = false;
+
+  //  Initial x&y coords.
+  private xCoord: number;
+  private yCoord: number;
 
   //  Create direct reference of canvas on editor.html
   constructor(
     private canvas: ElementRef<HTMLCanvasElement>, 
     private magicWandService: MagicWandService,
   ) { }
 
+/**
+ * Listens for user mousedown on 'appMask'. When user mouse goes down
+ * It initialized the set and paints the single pixel. 
```

Comment:
This second sentence doesn't quite make sense. I'd suggest: "Initializes the \`paintPixels\' set and paints a single pixel when the user presses the mouse."